pkg/build/api/v1/types.go

```diff
@@ -176,8 +176,89 @@ type BuildStatus struct {
 
 	// output describes the Docker image the build has produced.
 	Output BuildStatusOutput `json:"output,omitempty" protobuf:"bytes,10,opt,name=output"`
+
+	// steps contains details about each build step.
+	Stages Stages `json:"stages,omitempty" protobuf:"bytes,11,opt,name=stages"`
+}
+
+// Stages is a slice of build stages that have been recorded.
+type Stages []StageInfo
+
+// StageInfo contains details about a build stage.
+type StageInfo struct {
+	// StageName is the identifier for each build stage.
+	StageName StageName `json:"stageName,omitempty" protobuf:"bytes,1,opt,name=stageName"`
+
+	// StartTime identifies when this stage started.
+	StartTime unversioned.Time `json:"startTime,omitempty" protobuf:"bytes,2,opt,name=startTime"`
+
+	// Duration identifies how long this stage ran.
+	Duration unversioned.Duration `json:"duration,omitempty" protobuf:"bytes,3,opt,name=duration"`
+
+	// Steps contains details about each build step within a build stage.
+	Steps Steps `json:"steps,omitempty" protobuf:"bytes,4,opt,name=steps"`
 }
 
+// StageName is the identifier for each build stage.
+type StageName string
+
+// Valid StageNames
+const (
+	// StageFetch fetches the source.
+	StageFetchInputs StageName = "FetchInputs"
+
+	// StagePullImages pulls the docker images.
+	StagePullImages StageName = "PullImages"
+
+	// StageBuild builds the source.
+	StageBuild StageName = "Build"
+
+	// StagePost executes post steps.
+	StagePostCommit StageName = "PostCommit"
+
+	// StagePushImage pushes the docker image.
+	StagePushImage StageName = "PushImage"
+)
+
+// Steps is a slice of build steps that have been recorded
+type Steps []StepInfo
+
+// StepInfo contains details about a build step.
+type StepInfo struct {
+	// StepName is the identifier for each build step.
+	StepName StepName `json:"stepName,omitempty" protobuf:"bytes,1,opt,name=stepName"`
```

Comment:
`Name` instead of `StepName` unless we can make an argument that there will ever be a more basic "name" for a step that isn't this field.  Basically `build.status.steps[0].name` reads better than `build.status.steps[0].stepName`.